calculator/models.py

```diff
@@ -1,3 +1,143 @@
 from django.db import models
+from django.forms import DecimalField
+from django.core.validators import MinValueValidator
 
-# Create your models here.
+from decimal import Decimal, ROUND_DOWN
+from datetime import datetime, timezone
+
+
+def increment_loan_id():
+        id_loan = '{:015d}'.format(Loan.objects.count() + 1)
+        return '{}-{}-{}-{}'.format(id_loan[:3], id_loan[3:7], id_loan[7:11], id_loan[11:15])
+
+
+class Client(models.Model):
+    """
+    Client Model
+    Defines the attributes of a client
+    """
+
+    name = models.CharField("Name", max_length=30)
+    surname = models.CharField("Last name", max_length=30)
+    email = models.EmailField("E-mail")
+    phone = models.BigIntegerField("Phone")
+    cpf = models.BigIntegerField("CPF", unique=True)
+
+    @property
+    def is_indebted(self):
+        missed_payments = (
+            Payment.objects.filter(loan_id__client=self, status="missed")
+            .distinct()
+            .count()
+        )
+        if missed_payments >= 3:
+            return True
+        return False
+
+    class Meta:
+        verbose_name = "Client"
+        verbose_name_plural = "Clients"
+
+    def __str__(self):
+        return f"Client(id={self.id}, name={self.name}, surname={self.surname}, email={self.email}, phone={self.phone}, cpf={self.cpf})"
+
+
+class Loan(models.Model):
+    """
+    Loan Model
+    Defines the attributes of a loan
+    """
+    id = models.CharField(max_length=19, default=increment_loan_id, editable=False, primary_key=True)
+    client = models.ForeignKey(Client, on_delete=models.DO_NOTHING)
+    amount = models.DecimalField(
+        "Amount",
+        max_digits=15,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+    )
+    term = models.IntegerField("Term", validators=[MinValueValidator(1)])
+    rate = models.DecimalField(
+        "Rate",
+        max_digits=15,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+    )
+    date_initial = models.DateTimeField(
+        "Date creation", auto_now=False, auto_now_add=False
+    )
+    instalment = models.DecimalField(
+        "Instalment",
+        default=Decimal('0.0'),
+        max_digits=15,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+    )
+
+    def _instalment_adjustment(self):
+        loans_history = self.client.loan_set.all()
```

Comment:
Every time that you execute this function you will bring all loans, if you have millions of user maybe it is a bad ideia. use a cache should be better.